### ISSUE TITLE
No mapping for the Unicode character in target multi-byte code page

### ISSUE THREAD

**Issue by roughed** [HeidiSQL/HeidiSQL#1048]
I get this error when trying to load database
Size 1gb
Error comes at 180mb, goes through till 180mb with no errors

i selected utf8 as format when loading database. I've tried several other format as well to try if it works but doesn't load up properly.
I've uploaded the file on dropbox, here is the link of database reference

Can someone help please
https://www.dropbox.com/s/ma7qvzjw8ke8m4i/philly2philly.sql?dl=0

**Comment by ansgarbecker** [comment]
The link you provided gives me a 404/not-found error. Please repeat the upload if this is still a valid question.
Is there a reason why you deleted the whole issue template? Such issues will normally be just closed. Please ensure you are using the latest build and retry.

**Comment by caugner** [comment]
@ansgarbecker Would it be possible to provide more context in the error message, like some representation of the character as well as the line:column of that character in the imported file?

**Comment by ansgarbecker** [comment]
Yes, that's a good idea. That way it should be easier to find the offending characters and/or encoding issue.

**Comment by ansgarbecker** [comment]
Btw, is that one 8GB file you are trying to import, or several files? If the critical file is very large, how can we use an editor to view/check its contents? Notepad++ is nice, but does not support _such_ huge files.
We will definitely need a sample file with that issue, as otherwise I cannot test it.

**Comment by caugner** [comment]
> Btw, is that one 8GB file you are trying to import, or several files?

It's one file.

> If the critical file is very large, how can we use an editor to view/check its contents?

I would use `less` or `sed`, see: https://stackoverflow.com/a/83347

**Comment by caugner** [comment]
@ansgarbecker Would it be possible to show the beginning of the currently executed query in a third line of the `ProgressDialog`?

https://github.com/HeidiSQL/HeidiSQL/blob/a0fa16f3639e4a61fff2aebb6da268797d752a36/source/main.pas#L4010-L4020

This would make it probably much easier to find the line causing the issue.

PS: I tried to derive the causing line by looking at the partially imported database, but if I extract those lines with `sed -n 'FROM_LINE,TO_LINEp' FILE` (where TO_LINE is high enough to create a file causing the "execute directly" prompt), they import without problems.

**Comment by ansgarbecker** [comment]
As the [progress dialog](https://docs.microsoft.com/de-at/windows/win32/api/shlobj_core/nf-shlobj_core-iprogressdialog-setline) enables `PROGDLG_AUTOTIME`, the 3rd line is already in use for the automatic time display. But that's not an issue, I should also be able to display the offending text in the log panel.

**Comment by caugner** [comment]
Meanwhile, I have splitted the database dump into files per table using [mysqldumpsplitter](https://github.com/kedarvj/mysqldumpsplitter). These individual files seem to import without any issue, even if imported altogether.

So it looks to me like the queries themselves are not the issue, but rather HeidiSQL seems to stumble at some point and read the wrong bytes. Hopefully we'll see this once the offending text shows up.

I guess you're planning on logging the query (ideally only the beginning and the tail) in the following `except` section?
https://github.com/HeidiSQL/HeidiSQL/blob/a0fa16f3639e4a61fff2aebb6da268797d752a36/source/main.pas#L4024-L4034

PS: Not sure if it might help to make the chunk size configurable: https://github.com/HeidiSQL/HeidiSQL/blob/a0fa16f3639e4a61fff2aebb6da268797d752a36/source/main.pas#L3989

**Comment by caugner** [comment]
> Are you on the latest nightly Heidi build?

Yes, Nightly: 11.0.0.6096

**Comment by acqsk** [comment]
@ansgarbecker According to previous issue. Yes, you have truth, it can be problem inside sql file, but I'm sure, that I was selected the correct UTF-8 encoding before run SQL. File was created by adminer. Thanks a lot.

**Comment by thE-iNviNciblE** [comment]
> @ansgarbecker According to previous issue. Yes, you have truth, it can be problem inside sql file, but I'm sure, that I was selected the correct UTF-8 encoding before run SQL. File was created by adminer. Thanks a lot.

i've the same error with a nearly 10 GB Import-File.

**Comment by a-schild** [comment]
Same here with a 210mb restore from a wordpress dump in utf-8

```
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE `wp_posts` (
  `ID` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  `post_author` bigint(20) unsigned NOT NULL DEFAULT '0',
  `post_date` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
  `post_date_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',
  `post_content` longtext COLLATE utf8mb4_unicode_520_ci NOT NULL, 
```

**Comment by Dwza** [comment]
Same problem here.

My File is **~800MB** and it's exported in **UTF-8**
Import selection is UTF-8 as well.

My DB were I am Importing is a MariaDB 10.4.**28**-MariaDB
My file contains Server version 10.4.22-MariaDB

**Comment by ansgarbecker** [comment]
I would like to ask whether anyone is still able to reproduce the "_No mapping for the Unicode character_" error with a dump file?

I was never able to see the error, plus the sample file from the above DropBox link is offline.

**Comment by shadren** [comment]
Yes, but, no.
I ran into this issue on the latest build today (12.17.0.7303) with a 50MB dump file from MariaDB (MariaDB dump 10.19-11.4.9-MariaDB) generated on a Linux client. However, it was user error in the end. I'll continue with my details with the hope that is sheds light on the problem for others.

**TLDR:** The dump file was in **ANSI** encoding and the error occurred when trying to open it in HeidiSQL as UTF8, the default option.

The "fixed" file was created by copying the file from Notepad++ and pasting it into HeidiSQL and then copying it back to a new file in Notepad++, which was UTF8, then it worked. Then I use KDiff3 to see what was changed. Ignoring line endings, it was a few Unicode characters early on in the file in a blob content field. When I went back and checked in Notepadd++ I saw that the original file was ANSI encoded. I went back and tried the original file again in HeidiSQL, opening the file as ANSI and everything worked fine.

So nothing was wrong with HeidiSQL, just the users assumptions that the dump file was UTF8.

Sample dump file.

[HeidiSql-fail-import ANSI vs UTF8.sql](https://github.com/user-attachments/files/28567531/HeidiSql-fail-import.ANSI.vs.UTF8.sql)

**Comment by ansgarbecker** [comment]
Thanks for testing this.

I tried again to reproduce this time with the file you attached, and it works like expected without the reported error message.
